src/EditorFeatures/Core/DocumentationComments/CopilotGenerateDocumentationCommentProvider.cs

```diff
@@ -0,0 +1,333 @@
+                }
+
+                index = exceptionEndTag + "</exception>".Length;
+            }
+
+            return new DocumentationCommentProposal(memberNode.ToFullString(), proposedEdits.ToImmutableArray());
+        }
+
+        /// <summary>
+        /// Calls into the copilot service to get the pieces for the documentation comment.
+        /// </summary>
+        private static async Task<IReadOnlyList<ProposedEdit>> GetProposedEditsAsync(
+            DocumentationCommentProposal proposal, ICopilotCodeAnalysisService copilotService,
+            ITextSnapshot oldSnapshot, string? indentText, CancellationToken cancellationToken)
+        {
+            var list = new List<ProposedEdit>();
+            var (documentationCommentDictionary, isQuotaExceeded) = await copilotService.GetDocumentationCommentAsync(proposal, cancellationToken).ConfigureAwait(false);
+
+            // Quietly fail if the quota has been exceeded.
+            if (isQuotaExceeded)
+            {
+                return list;
+            }
+
+            if (documentationCommentDictionary is null)
+            {
+                return list;
+            }
+
+            if (documentationCommentDictionary.Count == 0)
+            {
+                return list;
+            }
+
+            foreach (var edit in proposal.ProposedEdits)
+            {
+                string? copilotStatement = null;
+                var textSpan = edit.SpanToReplace;
+
+                if (edit.TagType == DocumentationCommentTagType.Summary && documentationCommentDictionary.TryGetValue(DocumentationCommentTagType.Summary.ToString(), out var summary) && !string.IsNullOrEmpty(summary))
+                {
+                    copilotStatement = summary;
+                }
+                if (edit.TagType == DocumentationCommentTagType.TypeParam && documentationCommentDictionary.TryGetValue(edit.SymbolName!, out var typeParam) && !string.IsNullOrEmpty(typeParam))
+                {
+                    copilotStatement = typeParam;
+                }
+                else if (edit.TagType == DocumentationCommentTagType.Param && documentationCommentDictionary.TryGetValue(edit.SymbolName!, out var param) && !string.IsNullOrEmpty(param))
+                {
+                    copilotStatement = param;
+                }
+                else if (edit.TagType == DocumentationCommentTagType.Returns && documentationCommentDictionary.TryGetValue(DocumentationCommentTagType.Returns.ToString(), out var returns) && !string.IsNullOrEmpty(returns))
+                {
+                    copilotStatement = returns;
+                }
+                else if (edit.TagType == DocumentationCommentTagType.Exception && documentationCommentDictionary.TryGetValue(edit.SymbolName!, out var exception) && !string.IsNullOrEmpty(exception))
```

Comment:
while the chances are low, I guess it's possible that the there might be conflict in symbol names?